src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/StringTransferSources.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities.PooledObjects;
+using Microsoft.CodeAnalysis;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
+{
+    internal static class StringTranferSources
+    {
+        /// <summary>
+        /// <see cref="SourceInfo"/>s for transferring string tainted data sources.
+        /// </summary>
+        public static ImmutableHashSet<SourceInfo> SourceInfos { get; }
+
+        /// <summary>
+        /// Statically constructs.
+        /// </summary>
+        static StringTranferSources()
+        {
+            var sourceInfosBuilder = PooledHashSet<SourceInfo>.GetInstance();
+
+            sourceInfosBuilder.AddSourceInfoSpecifyingTaintedTargets(
+                WellKnownTypeNames.SystemTextStringBuilder,
+                isInterface: false,
+                taintedProperties: null,
+                taintedMethodsNeedsPointsToAnalysis: null,
+                taintedMethodsNeedsValueContentAnalysis: null,
+                transferProperties: new[] { "this[]" },
+                transferMethods: new (MethodMatcher, (string, string)[])[]{
+                    (
+                        (methodName, arguments) =>
+                            methodName == "Append" &&
+                            !arguments.IsEmpty &&
+                            (arguments[0].Parameter.Type.SpecialType == SpecialType.System_String ||
```

Comment:
Ah, you didn't **have** to consider parameter types here, because hopefully whatever is done by #4033 will solve it, but thanks for doing it anyway. :smile:

Reply:
The discussion in the issue is still valid. The change reduces the set of injectable sinks, but integers and other types are still tainted as a source and flow into the other sinks.